.circleci/config.yml

```diff
@@ -6,7 +6,7 @@ jobs:
   build:
     docker:
       # specify the version
-      - image: circleci/golang:1.14
+      - image: cimg/go:1.17.8
```

Comment:
This seems like a sensible change, but I'd prefer it as part of a separate pull-request. We should upgrade to go1.17 (or perhaps now go1.18) in one change and upgrade the circleci image in another.